dev/tests/integration/framework/Magento/TestFramework/Annotation/DbIsolation.php

```diff
@@ -49,6 +126,33 @@ public function endTestTransactionRequest(
     ) {
         if ($this->_isIsolationActive && $this->_getIsolation($test)) {
             $param->requestTransactionRollback();
+        } else {
+            $isolationProblem = [];
+            foreach (self::$isolationCache as $table => $isolationData) {
+                try {
+                    $diff = $this->dataDiff(
+                        $isolationData,
+                        $this->pullDbState($table)
+                    );
+
+                    if (!empty($diff)) {
+                        $isolationProblem[$table] = $diff;
+                    }
+                } catch (\Exception $e) {
+                    //ResourceConnection could be not specified in some specific tests that are not working with DB
+                    //We need to ignore it
+                }
+            }
+
+            if (!empty($isolationProblem)) {
```

Comment:
@skovalenk this exception raises BEFORE rollback transactions, so diff always not empty, that's why we have ALL integration tests as failed (over 8k)

Reply:
https://github.com/magento/magento2ce/pull/6036/files#diff-6f105113d42c6b5187894e275c4dab6f
moved this logic to class that can make db tables snapshot before test and compare with db tables after rollbacks

Reply:
@RuslanKostiv1 by default data fixture rollback should be triggered before transaction rollback, so if transaction rollback was triggered before data fixture rollback, than it is bug of integration framework and we need to fix it there.
Please see: \Magento\TestFramework\EventManager::fireEvent

Reply:
@skovalenk this exception raises BEFORE rollback transactions, so diff always not empty, that's why we have ALL integration tests as failed (over 8k)

I ran this only for tests with dbIsolation disabled. I don`t think we even have 8k of tests with dbIsolation disabled.

Reply:
@skovalenk , 8,138 failures

please debug your changes with any test and you'll see that \Magento\TestFramework\Annotation\DbIsolation.php::endTestTransactionRequest (with data comparison) will be executed before \Magento\TestFramework\Annotation\AbstractDataFixture::_revertFixtures. I've moved all these changes to \Magento\TestFramework\Annotation\DataFixture